amethyst_utils/src/auto_fov.rs

```diff
@@ -19,35 +19,38 @@ use serde::{Deserialize, Serialize};
 use thread_profiler::profile_scope;
 
 /// A component describing the behavior of the camera in accordance with the screen dimensions
-#[derive(Clone, Debug, Deserialize, PrefabData, Serialize)]
+#[derive(Clone, Debug, Deserialize, PrefabData, Serialize, new, Setters)]
 #[prefab(Component)]
 #[serde(default)]
 pub struct AutoFov {
     /// The horizontal FOV value at the aspect ratio in the field `base_aspect_ratio`
+    #[new(default)]
     base_fovx: f32,
 
+    /// Sets `fovx_growth_rate` to the given value
+    #[set = "pub"]
     /// The factor determining how sensitive the FOV change should be
+    #[new(default)]
     fovx_growth_rate: f32,
 
     /// If the FOV grow rate specified in the field `fovx_growth_rate` should be applied as-is
+    #[new(default)]
     fixed_growth_rate: bool,
```

Comment:
I think you forgot to add the get/set macros for those?

Reply:
They have non-trivial setters and getters that return by value, so I can't add macros for those